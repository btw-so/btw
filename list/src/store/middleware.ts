import createSagaMiddleware from 'redux-saga';

export const sagaMiddleware = createSagaMiddleware();

const middleware = [sagaMiddleware];

/* istanbul ignore next */
if (process.env.NODE_ENV === 'development') {
  const { createLogger } = require('redux-logger');
  const invariant = require('redux-immutable-state-invariant').default;

  middleware.push(invariant(), createLogger({ collapsed: true }));
}

export default middleware;
