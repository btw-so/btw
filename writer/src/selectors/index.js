import { createSelector } from "@reduxjs/toolkit";

export const selectApp = createSelector(
  (state) => state.app,
  (app) => app
);

export const selectGitHub = createSelector(
  (state) => state.github,
  (github) => github
);

export const selectUser = createSelector(
  (state) => state.user,
  (user) => user
);

export const selectNotes = createSelector(
  (state) => state.notes,
  (notes) => notes
);

export const selectNoteActions = createSelector(
  (state) => state.actions,
  (actions) => actions
);

export const selectOtp = createSelector(
  (state) => state.otp,
  (otp) => otp
);

export const selectList = createSelector(
  (state) => state.list,
  (list) => list
);
