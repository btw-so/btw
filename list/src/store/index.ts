import {
  applyMiddleware,
  compose,
  createStore,
  Middleware,
  Store,
} from "redux";
import { persistCombineReducers, persistStore } from "redux-persist";
import autoMergeLevel2 from "redux-persist/lib/stateReconciler/autoMergeLevel2";
import storage from "redux-persist/lib/storage";

import reducers from "reducers";
import rootSaga from "sagas";

import { RootState } from "types";

import middleware, { sagaMiddleware } from "./middleware";

const rootReducer = persistCombineReducers<RootState>(
  {
    key: "rrsb",
    stateReconciler: autoMergeLevel2,
    storage,
    blacklist: ["otp", "actions"],
    timeout: 0,
    version: 1,
  },
  reducers
);

const composeEnhancer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

/* istanbul ignore next */
export const configStore = (
  initialState: any = {},
  additionalMiddleware: Middleware[] = []
) => {
  const store: Store = createStore(
    rootReducer,
    initialState,
    composeEnhancer(applyMiddleware(...additionalMiddleware, ...middleware))
  );

  sagaMiddleware.run(rootSaga);

  if (module.hot) {
    module.hot.accept("reducers", () => {
      store.replaceReducer(rootReducer);
    });
  }

  return {
    persistor: persistStore(store),
    store,
  };
};
