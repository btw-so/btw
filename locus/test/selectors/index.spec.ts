import { selectApp, selectGitHub, selectUser } from 'selectors';
import { configStore } from 'store';

describe('selectors', () => {
  const { store } = configStore();

  describe('selectApp', () => {
    it('should return the app', () => {
      expect(selectApp(store.getState())).toMatchSnapshot();
    });
  });

  describe('selectGitHub', () => {
    it('should return the github', () => {
      expect(selectGitHub(store.getState())).toMatchSnapshot();
    });
  });

  describe('selectUser', () => {
    it('should return the user', () => {
      expect(selectUser(store.getState())).toMatchSnapshot();
    });
  });
});
