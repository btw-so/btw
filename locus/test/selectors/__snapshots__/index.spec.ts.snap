// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`selectors selectApp should return the app 1`] = `
Object {
  "query": "react",
}
`;

exports[`selectors selectGitHub should return the github 1`] = `
Object {
  "topics": Object {},
}
`;

exports[`selectors selectUser should return the user 1`] = `
Object {
  "isAuthenticated": false,
  "status": "idle",
}
`;
